src/Microsoft.ML.TimeSeries/SrCnnEntireTransformer.cs

```diff
@@ -0,0 +1,198 @@
+                OutputLength = 7;
+            }
+        }
+
+        private SrCnnEntireTransformer(IHost host, ModelLoadContext ctx)
+            : base(host, ctx)
+        {
+            //TODO
+        }
+
+        private static IDataTransform Create(IHostEnvironment env, Options options, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(options, nameof(options));
+            env.CheckValue(input, nameof(input));
+
+            return new SrCnnEntireTransformer(env, options, input).MakeDataTransform(input);
+        }
+
+        private static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register(LoaderSignature);
+
+            host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+
+            return new SrCnnEntireTransformer(host, ctx).MakeDataTransform(input);
+        }
+
+        internal static SrCnnEntireTransformer Create(IHostEnvironment env, ModelLoadContext ctx)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register(LoaderSignature);
+
+            host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+
+            return new SrCnnEntireTransformer(host, ctx);
+        }
+
+        private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, DataViewSchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        private protected override void SaveModel(ModelSaveContext ctx)
+        {
+            //TO DO
+            throw new NotImplementedException();
+        }
+
+        private protected override IRowMapper MakeRowMapper(DataViewSchema schema) => new Mapper(this, schema);
+
+        private sealed class Mapper : OneToOneMapperBase
+        {
+            private readonly SrCnnEntireTransformer _parent;
+            private readonly VBuffer<ReadOnlyMemory<Char>> _slotNames;
+
+            public Mapper(SrCnnEntireTransformer parent, DataViewSchema inputSchema)
+                : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                _parent = parent;
+                if (_parent.OutputLength == 3)
+                {
+                    _slotNames = new VBuffer<ReadOnlyMemory<char>>(_parent.OutputLength, new[] { "Alert".AsMemory(), "Raw Score".AsMemory(), "Mag".AsMemory() });
+                }
+                else if (_parent.OutputLength == 7)
+                {
+                    _slotNames = new VBuffer<ReadOnlyMemory<char>>(_parent.OutputLength, new[] { "Is Anomaly".AsMemory(), "Anomaly Score".AsMemory(), "Mag".AsMemory(),
+                        "Expected Value".AsMemory(), "Boundary Unit".AsMemory(), "Upper Boundary".AsMemory(), "Lower Boundary".AsMemory() });
+                }
+                //TODO
+            }
```

Comment:
Can you please add more details on exactly what the TODO work is?  Or delete them if they are already done.

Reply:
> Can you please add more details on exactly what the TODO work is? Or delete them if they are already done.

Deleted.